presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -423,6 +423,9 @@
     private final HiveEncryptionInformationProvider encryptionInformationProvider;
     private final HivePartitionStats hivePartitionStats;
     private final HiveFileRenamer hiveFileRenamer;
+    // Do not remove this. HiveMetadata is used as a library outside the Presto repository.
+    // The outside library sets this to true to control the behavior of the HiveMetadata.
```

Comment:
It seems awkward to have this in the code - but no way to set/change the value. I wonder if it would make sense to derive a version of HiveMetadata with a different behavior, or maybe add a configuration property to toggle this field?

Reply:
I believe the previous one was via HiveTypeTranslator, we had this code to derive the behavior. 

HiveMetadataFactory took in a HiveTypeTranslator to control the behavior. Now instead of the polymorphism it uses the feature flag to control the behavior. There is test to control the testing.

Thoughts ?

Reply:
Other way is to put the TypeTranslator back. It would have been nice, if the previous PR was one a refactoring change and the other a feature addition change, it would have been easier to revert the refactor.

Reply:
@ClarenceThreepwood  Let me know if you want to proceed with reverting the refactor. I am comfortable going forward with this change, as it is a functionality controlled by a feature flag and there are tests to cover this behavior.

Reply:
ok. I'm fine with going ahead. This was a minor reservation on my end